RealmSwift/ThreadSafeReference.swift

```diff
@@ -122,6 +122,78 @@ public protocol ThreadConfined {
     }
 }
 
+/**
+ Object types conforming to `ThreadSafeWrappable` are able to have the `@ThreadSafe` property wrapper applied to them.
+
+ Note that only types defined by Realm can meaningfully conform to this protocol, and defining new
+ classes which attempt to conform to it will not make them work with `@ThreadSafe`
+ */
+public protocol ThreadSafeWrappable: ThreadConfined {}
+/**
+    A property wrapper type that may be passed between threads.
+
+    A `@ThreadSafe` property contains a thread-safe reference to the underlying wrapped value.
+    This reference is resolved to the thread on which the wrapped value is accessed. A new thread
+    safe reference is created each time the property is accessed.
+
+ - warning: This property wrapper should not be used for properties on long lived objects.
+            `@ThreadSafe` properties contain a `ThreadSafeReference` which
+            can pin the source version of the Realm in use. This means that this property
+            wrapper is **better suited for function arguments and local variables**
+            **that get captured by an aynchronously dispatched block.**
+
+ - see: `ThreadSafeReference`
+ - see: `ThreadConfined`
+*/
+@propertyWrapper public class ThreadSafe<T: ThreadSafeWrappable> {
+    var threadSafeReference: ThreadSafeReference<T>?
+    var rlmConfiguration: RLMRealmConfiguration?
+
+    /// :nodoc:
+    public var wrappedValue: T? {
+        get {
+            guard let threadSafeReference = threadSafeReference, let rlmConfig = rlmConfiguration else { return nil }
+            do {
+                let rlmRealm = try RLMRealm(configuration: rlmConfig)
+                let realm = Realm(rlmRealm)
+                guard let value = threadSafeReference.resolve(in: realm) else {
+                    self.threadSafeReference = nil
+                    return nil
+                }
+                self.threadSafeReference = ThreadSafeReference(to: value)
+                return value
+            // FIXME: wrappedValue should throw
+            // As of Swift 5.5 property wrappers can't have throwing accessors.
+            } catch let error as NSError { throwRealmException(error.localizedDescription) }
+        }
+        set {
+            guard let newValue = newValue else {
+                threadSafeReference = nil
+                return
+            }
+            guard let configuration = newValue.realm?.configuration else {
```

Comment:
`newValue.realm?.rlmRealm.configuration`

Currently this is converting the RLMRealmConfiguration to a Realm.Configuration and then back, and we want to instead grab the RLMRealmConfiguration from the RLMRealm directly.